clustered/server/src/test/java/org/ehcache/clustered/server/store/ServerStoreTest.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.store;
+
+
+import org.ehcache.clustered.common.store.Chain;
+import org.ehcache.clustered.common.store.ChainBuilder;
+import org.ehcache.clustered.common.store.Element;
+import org.ehcache.clustered.common.store.ElementBuilder;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Verify Server Store
+ */
+public abstract class ServerStoreTest {
+
+  public abstract ServerStore newStore();
+
+  public abstract ChainBuilder newChainBuilder();
+
+  public abstract ElementBuilder newElementBuilder();
+
+  private final ChainBuilder chainBuilder = newChainBuilder();
+  private final ElementBuilder elementBuilder = newElementBuilder();
+
+  private Element compactChain(Chain chain, ByteBuffer payLoad) {
+
+  Iterator<Element> elements = chain.reverseIterator();
+  Element toUpdate = elements.next();
+
+  return elementBuilder.getElement(toUpdate, payLoad);
+  }
+
+  private static void populateStore(ServerStore store) {
+    for(int i = 1 ; i <= 16; i++) {
+      store.append(i, getPayload(i));
+    }
+  }
+
+  private static long readPayLoad(ByteBuffer byteBuffer) {
+    byteBuffer.flip();
+    return byteBuffer.getLong();
+  }
+
+  private static ByteBuffer getPayload(long key) {
+    ByteBuffer byteBuffer = ByteBuffer.allocate(8).putLong(key);
```

Comment:
You either need to do an absolute put here, or flip the buffer before returning it.  At the moment this is returning buffers with zero remaining bytes, so they'll all be equal to each other.